flake.nix

```
@@ -1,34 +1,54 @@
 {
   inputs.flake-utils.url = github:numtide/flake-utils;
 
-  nixConfig = {
-    # Needed by callCabal2nix
-    allow-import-from-derivation = true;
-  };
-
   outputs = {
     self,
     nixpkgs,
     flake-utils,
   }:
     with flake-utils.lib;
-      eachSystem [system.x86_64-linux] (system: let
+      eachDefaultSystem (system: let
         pkgs = nixpkgs.legacyPackages.${system};
         hpkgs = pkgs.haskellPackages;
-        smtlib-backends = hpkgs.callCabal2nix "smtlib-backends" ./. {};
-        smtlib-backends-process = hpkgs.callCabal2nix "smtlib-backends-process" ./smtlib-backends-process {inherit smtlib-backends smtlib-backends-tests;};
-        smtlib-backends-tests = hpkgs.callCabal2nix "smtlib-backends-tests" ./smtlib-backends-tests {inherit smtlib-backends;};
-        smtlib-backends-z3 = hpkgs.callCabal2nix "smtlib-backends-z3" ./smtlib-backends-z3 {
+        regen = pkgs.writeShellApplication {
+          name = "regen";
+          ## coreutils for xargs
+          runtimeInputs = with pkgs; [cabal2nix findutils coreutils];
+          text = ''
+            set -o errexit
+            set -o pipefail
+            set -o xtrace
+            ## Nix automatically runs `shellcheck` on this script
+            # shellcheck disable=SC2016
+            find . -name '*.cabal' -print0 | \
+            xargs --null -I{} \
+            sh -c 'cabal2nix "$(dirname {})" > "$(basename --suffix='.cabal' {}).nix"'
```

Is there a way to indent Shell in the string to make it more readable or will Alejandra complain? For instance:
```suggestion
            find . -name '*.cabal' -print0 | \
                xargs --null -I{} \
                    sh -c 'cabal2nix "$(dirname {})" > "$(basename --suffix='.cabal' {}).nix"'
```